h2/src/main/org/h2/jdbc/JdbcBlob.java

```diff
@@ -129,6 +132,9 @@ public int setBytes(long pos, byte[] bytes) throws SQLException {
     @Override
     public int setBytes(long pos, byte[] bytes, int offset, int len)
             throws SQLException {
+        if (bytes == null) {
+            return 0;
```

Comment:
I mean that `setBytes(1, null, 0, 1)`, `setBytes(1, null, 1, 0)`, etc. should throw some exception (NPE or other) instead of returning 0 to indicate a problem.

Reply:
Also I think that invocation with invalid or unsupported position (H2 allows only 1) should not complete normally even if data is null or empty.

Reply:
Agreed on both counts